### ISSUE TITLE
Whisper support in Windows for Youtube Videos

### ISSUE THREAD

**Issue by cherrerajobs** [h2oai/h2ogpt#1155]
I'm running windows 10 and setup h2ogpt via the Windows Installation MD file here:
https://github.com/h2oai/h2ogpt/blob/main/docs/README_WINDOWS.md

I confirm Text to Speech and Speech to Text working in Windows.

However I get the following output when attempting to transcribe a youtube video via ingestion:
```
load INSTRUCTOR_Transformer
max_seq_length  512
Downloading config.json: 100%|█████████████████████████████████████████████████████████| 1.94k/1.94k [00:00<?, ?B/s]
Downloading model.safetensors: 100%|█████████████████████████████████████████████| 290M/290M [00:39<00:00, 7.44MB/s]
Downloading generation_config.json: 100%|██████████████████████████████████████████████| 1.50k/1.50k [00:00<?, ?B/s]
Downloading tokenizer_config.json: 100%|███████████████████████████████████████████████████| 805/805 [00:00<?, ?B/s]
Downloading vocab.json: 100%|████████████████████████████████████████████████████| 798k/798k [00:00<00:00, 3.73MB/s]
Downloading tokenizer.json: 100%|██████████████████████████████████████████████| 2.41M/2.41M [00:00<00:00, 6.16MB/s]
Downloading merges.txt: 100%|████████████████████████████████████████████████████| 456k/456k [00:00<00:00, 12.3MB/s]
Downloading normalizer.json: 100%|██████████████████████████████████████████████| 52.7k/52.7k [00:00<00:00, 785kB/s]
Downloading added_tokens.json: 100%|███████████████████████████████████████████| 34.6k/34.6k [00:00<00:00, 34.6MB/s]
Downloading (…)cial_tokens_map.json: 100%|█████████████████████████████████████████████| 1.83k/1.83k [00:00<?, ?B/s]
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
Downloading (…)rocessor_config.json: 100%|███████████████████████████████████████| 185k/185k [00:00<00:00, 13.2MB/s]
Running on local URL:  http://0.0.0.0:7860

To create a public link, set `share=True` in `launch()`.
Started Gradio Server and/or GUI: server_name: localhost port: None
Use local URL: http://localhost:7860/
WARNING! Model override. Might not fit in your GPU
Using the following model:  openai/whisper-medium
Downloading config.json: 100%|█████████████████████████████████████████████████████████| 1.99k/1.99k [00:00<?, ?B/s]
Downloading model.safetensors: 100%|███████████████████████████████████████████| 3.06G/3.06G [06:48<00:00, 7.47MB/s]
Downloading generation_config.json: 100%|██████████████████████████████████████████████| 3.69k/3.69k [00:00<?, ?B/s]
Downloading tokenizer_config.json: 100%|████████████████████████████████████████████| 805/805 [00:00<00:00, 804kB/s]
Downloading vocab.json: 100%|████████████████████████████████████████████████████| 836k/836k [00:00<00:00, 4.02MB/s]
Downloading tokenizer.json: 100%|██████████████████████████████████████████████| 2.48M/2.48M [00:00<00:00, 6.58MB/s]
Downloading merges.txt: 100%|████████████████████████████████████████████████████| 494k/494k [00:00<00:00, 12.3MB/s]
Downloading normalizer.json: 100%|█████████████████████████████████████████████| 52.7k/52.7k [00:00<00:00, 17.5MB/s]
Downloading added_tokens.json: 100%|███████████████████████████████████████████| 34.6k/34.6k [00:00<00:00, 17.3MB/s]
Downloading (…)cial_tokens_map.json: 100%|█████████████████████████████████████████████| 2.08k/2.08k [00:00<?, ?B/s]
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
Downloading (…)rocessor_config.json: 100%|███████████████████████████████████████| 185k/185k [00:00<00:00, 13.2MB/s]
The BetterTransformer implementation does not support padding during training, as the fused kernels do not support attention masks. Beware that passing padded batched data during training may result in unexpected outputs. Please refer to https://huggingface.co/docs/optimum/bettertransformer/overview for more details.
[youtube] Extracting URL: https://www.youtube.com/shorts/MN9vsZs7jkQ
[youtube] MN9vsZs7jkQ: Downloading webpage
[youtube] MN9vsZs7jkQ: Downloading ios player API JSON
[youtube] MN9vsZs7jkQ: Downloading android player API JSON
[youtube] MN9vsZs7jkQ: Downloading m3u8 information
[info] MN9vsZs7jkQ: Downloading 1 format(s): 140
[download] Destination: tmp\_7998ec40-f\YOUR ЯOЯЯIM IMAGE.m4a
[download] 100% of  950.51KiB in 00:00:00 at 4.88MiB/s
[FixupM4a] Correcting container of "tmp\_7998ec40-f\YOUR ЯOЯЯIM IMAGE.m4a"
[ExtractAudio] Not converting audio tmp\_7998ec40-f\YOUR ЯOЯЯIM IMAGE.m4a; file is already in target format m4a
```

Basically, the loader animation in gradio quits after that last `[ExtractAudio]` output and it shows 0 documents imported. No error.
![image](https://github.com/h2oai/h2ogpt/assets/8894763/a6703cfe-bd00-4ed0-b0bc-34c54fae3dfc)

Further attempts look the same except without model downloads.

**Comment by pseudotensor** [comment]
I haven't tried windows myself yet, will need to build one-click installer within next week and will see if any issues.

After the above it should say  more.  i.e.

```
Running on local URL:  http://0.0.0.0:7860

To create a public link, set `share=True` in `launch()`.
Started Gradio Server and/or GUI: server_name: localhost port: None
Use local URL: http://localhost:7860/
Adding ['https://www.youtube.com/shorts/MN9vsZs7jkQ']
BEGIN Consuming path_or_paths=None url=['https://www.youtube.com/shorts/MN9vsZs7jkQ'] text=['https://www.youtube.com/shorts/MN9vsZs7jkQ']
Ingesting URL or Text: https://www.youtube.com/shorts/MN9vsZs7jkQ
Fresh ASR
WARNING! Model override. Might not fit in your GPU
Using the following model:  openai/whisper-medium
  0%|          | 0/1 [00:00<?, ?it/s]Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
The BetterTransformer implementation does not support padding during training, as the fused kernels do not support attention masks. Beware that passing padded batched data during training may result in unexpected outputs. Please refer to https://huggingface.co/docs/optimum/bettertransformer/overview for more details.
[youtube] Extracting URL: https://www.youtube.com/shorts/MN9vsZs7jkQ
[youtube] MN9vsZs7jkQ: Downloading webpage
[youtube] MN9vsZs7jkQ: Downloading ios player API JSON
[youtube] MN9vsZs7jkQ: Downloading android player API JSON
[youtube] MN9vsZs7jkQ: Downloading m3u8 information
[info] MN9vsZs7jkQ: Downloading 1 format(s): 140
[download] Destination: /tmp/_d2589e48-1/YOUR ЯOЯЯIM IMAGE.m4a
[download] 100% of  950.51KiB in 00:00:00 at 5.09MiB/s   
[FixupM4a] Correcting container of "/tmp/_d2589e48-1/YOUR ЯOЯЯIM IMAGE.m4a"
[ExtractAudio] Not converting audio /tmp/_d2589e48-1/YOUR ЯOЯЯIM IMAGE.m4a; file is already in target format m4a
Transcribing part /tmp/_d2589e48-1/YOUR ЯOЯЯIM IMAGE.m4a!
100%|██████████| 1/1 [00:10<00:00, 10.20s/it]
0it [00:00, ?it/s]
DONE Ingesting URL or Text: https://www.youtube.com/shorts/MN9vsZs7jkQ
END consuming path_or_paths=None url=['https://www.youtube.com/shorts/MN9vsZs7jkQ'] text=['https://www.youtube.com/shorts/MN9vsZs7jkQ']
load INSTRUCTOR_Transformer
max_seq_length  512
```

Total time was about 10 seconds on my home pc with 3090ti.

**Comment by cherrerajobs** [comment]
I was able to fix that issue with pull request #1157 on Windows.
![image](https://github.com/h2oai/h2ogpt/assets/8894763/04d55b90-9e11-4a0e-886c-5d2bb9dc6cdc)
